### PR TITLE
Fix arrange/arrange_nodes methods filtering out some nodes.

### PR DESCRIPTION
**Pull request by trafium**
https://github.com/stefankroes/ancestry/issues/362

### CODE REVIEW

**Comment by vzharkov** [comment]
@trafium this PR break the ordering for the `arrange` method.

File: lib/ancestry/class_methods.rb [hunk 1]
```diff
@@ -29,27 +29,25 @@ def orphan_strategy= orphan_strategy
...
+      arrange_nodes self.where(options)
     end
 
     # Arrange array of nodes into a nested hash of the form
     # {node => children}, where children = {} if the node has no children
     def arrange_nodes(nodes)
-      arranged = ActiveSupport::OrderedHash.new
-      min_depth = Float::INFINITY
-      index = Hash.new { |h, k| h[k] = ActiveSupport::OrderedHash.new }
+      arranged  = ActiveSupport::OrderedHash.new
```
Comment (kbrock) [inline lib/ancestry/class_methods.rb]: hashes are ordered by default. can we just use a regular hash?

File: lib/ancestry/class_methods.rb [hunk 2]
```diff
@@ -29,27 +29,25 @@ def orphan_strategy= orphan_strategy
 
     # Arrangement
     def arrange options = {}
-      # Get all nodes ordered by ancestry and start sorting them into an empty hash
-      arrange_nodes self.ancestry_base_class.reorder(options.delete(:order)).where(options)
```
Comment (kbrock) [inline lib/ancestry/class_methods.rb]: not sure how removing the ordering helps the pr

The issue with removing this ordering is the tree will still get created, but the nodes will not be in the secondary desired order.

Also of note, the ancestry column does not have `c` collation. So I do not trust sql's `order` right now.

**Comment by kbrock** [comment]
Thank you for opening this PR.

future suggestion: make a local branch rather than creating this on your master.

I cant merge without a test, or with it breaking existing tests.
I also don't know how we are to figure out how to handle the missing 

Please reopen / ping me if you want to work through this

**Comment by kbrock** [comment]
@trafium so sorry for closing this.

I pulled this change (a little tweaked) into #415

**Pull request closed without merging**